spec/unit/tariff_synchronizer/base_update_spec.rb

```diff
@@ -110,26 +120,29 @@
       create :cds_update, :applied, issue_date: 1.day.ago.to_date
 
       (20.days.ago.to_date..Time.zone.today).each do |download_date|
-        expect(TariffSynchronizer::CdsUpdateDownloader).to receive(:new).with(download_date).and_return(instance_double('TariffSynchronizer::CdsUpdateDownloader', perform: nil))
+        allow(TariffSynchronizer::CdsUpdateDownloader).to receive(:new).with(download_date).and_return(instance_double('TariffSynchronizer::CdsUpdateDownloader', perform: nil))
       end
 
-      TariffSynchronizer::CdsUpdate.sync
+      TariffSynchronizer::CdsUpdate.sync(initial_date: 20.days.ago.to_date)
+
+      (20.days.ago.to_date..Time.zone.today).each do |download_date|
+        expect(TariffSynchronizer::CdsUpdateDownloader).to have_received(:new).with(download_date)
+      end
     end
   end
 
   describe '#last_updates_are_missing?' do
     context 'with weekends' do
       before do
         travel_to Date.parse('21-05-2017')
+        @update1 = create(:taric_update, :missing, example_date: Time.zone.today)
```

Comment:
update1 and update2 aren't meaningfully named:

```ruby
missing_update = create :taric_update, :missing, example_date: Time.zone.today
pending_update = create :taric_update, example_date: Time.zone.yesterday
```

Also, instance variables in tests are frowned on generally https://www.betterspecs.org/

Reply:
Missing updates are not actually possible any more so we can actually probably just remove any tests of missing updates